### PR TITLE
colin/refactor #288 custom hooks

### PR DESCRIPTION
**Pull request by colinvandergraaf**
- **Add custom-hook audit decsion doc**
- **Integrate useSignOut into useAuth hook**
- **WIP: Begin refactor of useAuth to include UserProfile check**

## Description
This PR adds a decision document for an audit of custom hooks and includes a refactor of the useAuth hook to include useSignout functionality.   

It is intended to include a check for UserProfile to address the case where a user can be authenticated with Firebase but not have a profile and be logged in,  but this is currently blocked by #356.

Fixes # (issue)
#288 

## Type of change

- [x]  🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨  New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Can this be tested? Testing Plan to review this PR

N/A
- [ ] Where in the app does this change take place?
- [ ] What are the steps to test this change?
- [ ] What is the expected output?

## Checklist:

- [ ] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

### CODE REVIEW

**Review by tedgusek — COMMENTED** [review]
## Line 13: Remove the word 'be'

Custom hooks are a mechansim for abstracting and encapsulating business logic and functionality to (be) share across components within a React applicaiton...

## Line 16: add 'it'

When is 'it' a good time...

## Everything else looks great!  

Thank you, I got a lot out of reading this

File: client/components/layout/MainLayout.tsx
Comment (tedgusek) [inline client/components/layout/MainLayout.tsx:1]: Looks great! 
Question about the console.errors, were these left in intentionally?  If so, cool, no further comments. If not, would this be a good place to utilize the snackbar to throw an error?

File: client/hooks/useAuth.ts
Comment (tedgusek) [inline client/hooks/useAuth.ts:1]: Looks good

**Review by tedgusek — COMMENTED** [review]
Overall looks great, just a couple minor non breaking details to look at.

File: .github/decisions/Custom_Hooks_Audit.md
Comment (LucieSeidler) [inline .github/decisions/Custom_Hooks_Audit.md:1]: I think this file could've potentially had it's own PR for documentation but thank you for writing it.

File: client/hooks/useAuth.ts [hunk 1]
```diff
@@ -3,30 +3,52 @@
...
+import type { FirebaseError } from 'firebase/app';
+import useFetchUserProfile from './useFetchUserProfile';
 
 const useAuth = () => {
   const [isAuthenticated, setIsAuthenticated] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
+  const queryClient = useQueryClient();
+  const { refetch } = useFetchUserProfile();
+
+  const signOutMutation = useMutation<void, FirebaseError, void>({
```
Comment (LucieSeidler) [inline client/hooks/useAuth.ts:22]: thank you for making our code more concise by joining the two auth hooks into one + a mutation

File: .github/decisions/Custom_Hooks_Audit.md
Comment (LucieSeidler) [reply .github/decisions/Custom_Hooks_Audit.md:1]: once ted's minor comments are addressed, I'll approve. Also please finish filling out the PR form

**Comment by relandboyle** [comment]
Closing due to inactivity. Will open a new PR when the issue is reassigned.

**Pull request closed without merging**